### ISSUE TITLE
zld+macos: handle weak frameworks

### ISSUE THREAD

**Issue by kubkon** [ziglang/zig#10206]
### Zig Version

0.9.0-dev.1703+3eeb468ad

### Steps to Reproduce

`zig c++ test.cpp -Wl,-weak_framework,Metal -Wl,-weak_framework,QuartzCore`

### Expected Behavior

No warnings, and flags respected.

### Actual Behavior

```
warning: unsupported linker arg: -weak_framework
warning: unsupported linker arg: Metal
warning: unsupported linker arg: -weak_framework
warning: unsupported linker arg: QuartzCore
```

**Comment by andrewrk** [comment]
Oops, moved the milestone because I mistakenly thought this was solved by #10208. But I understand that this is not fully solved yet, correct?

**Comment by kubkon** [comment]
> Oops, moved the milestone because I mistakenly thought this was solved by #10208. But I understand that this is not fully solved yet, correct?

Correct! I still need to work out exactly how to handle/denote weak imports in zld.
